### DISCUSSION TITLE
WslRegisterDistribution failed with error: 0x803b0001

### DISCUSSION THREAD

**Discussion by TomHebert** [microsoft/WSL#7270] [Q&A]
I am using Kali and I've tried to use Ubuntu 20.xx.  Kali was successfully being run on WSL.  In the process of trying to passthrough a USB device to Kali, I changed the networking device to the USB device and that is when I began to receive the error in the title.

I cannot find any documentation on the error.

I have tried to uncheck the WSL feature, restart, and recheck the feature to start all over and continue to receive the same error code.  I have also reconfigured the virtual switch manager to the default settings.

**Comment by meback** [comment]
如果你再次在WSL中安装ubuntu，启动显示错误 WslRegisterDistribution failed with error:0x803b0001
极有可能你是修改了虚拟交换机造成的。将虚拟交互机中WSL交换机删除
用管理员模式启动powershell执行
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All
Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
重新安装ubuntu或启动即可
